packages/Chem/detectors.js

```diff
@@ -18,21 +21,47 @@ class ChemPackageDetectors extends DG.Package {
   detectRDSmiles(col) {
     if (col.type !== DG.TYPE.STRING)
       return null;
-
-    let name = col.name.toLowerCase();
-    if (ChemPackageDetectors.likelyNames.some((likelyName) => name.endsWith(likelyName))) {
-      col.semType = DG.SEMTYPE.MOLECULE;
-
-      // smiles or molblock?
-      let str = col.length > 0 ? col.get(0) : null;
-      col.tags[DG.TAGS.UNITS] = (str !== null && str.includes('M  END')) ? DG.UNITS.Molecule.MOLBLOCK : DG.UNITS.Molecule.SMILES;
-
-      return col.semType;
+    // return DG.SEMTYPE.MOLECULE;
+    for (let i = 0; i < col.length; ++i) {
+      const el = col.get(i);
+      if (el && el.length > 0) {
+        if (!ChemPackageDetectors.validFirstChar(el[0]))
+          return null;
+        for (let j = 1; j < el.length; ++j)
+          if (!ChemPackageDetectors.validChar(el[j]))
+            return null;
+      }
     }
 
-    if (DG.Detector.sampleCategories(col, (s) => ChemPackageDetectors.likelySmiles(s)))
-      return DG.SEMTYPE.MOLECULE;
+    let molStrings = [];
+    if (col.length <= ChemPackageDetectors.numberOfMolsToCheck)
+      molStrings = col.toList().filter(el => el);
+    else {
+      const inc = Math.floor(col.length / ChemPackageDetectors.numberOfMolsToCheck);
+      for (let i = 0; i < col.length; i += inc) {
```

Comment:
You are iterating to col.length and not to inc

Reply:
Oh, I see, you were doing the check in the other if branch. Try to simplify the code